core/primitives-core/src/profile.rs

```diff
@@ -19,154 +58,464 @@ impl Default for ProfileData {
+                &mut self.data[50]
+            }
+            Cost::ExtCost { ext_cost_kind: ExtCosts::storage_iter_create_from_byte } => {
+                &mut self.data[51]
+            }
+            Cost::ExtCost { ext_cost_kind: ExtCosts::storage_iter_create_to_byte } => {
+                &mut self.data[52]
+            }
+            Cost::ExtCost { ext_cost_kind: ExtCosts::storage_iter_next_base } => &mut self.data[53],
+            Cost::ExtCost { ext_cost_kind: ExtCosts::storage_iter_next_key_byte } => {
+                &mut self.data[54]
+            }
+            Cost::ExtCost { ext_cost_kind: ExtCosts::storage_iter_next_value_byte } => {
+                &mut self.data[55]
+            }
+            Cost::ExtCost { ext_cost_kind: ExtCosts::touching_trie_node } => &mut self.data[56],
+            Cost::ExtCost { ext_cost_kind: ExtCosts::promise_and_base } => &mut self.data[57],
+            Cost::ExtCost { ext_cost_kind: ExtCosts::promise_and_per_promise } => {
+                &mut self.data[58]
+            }
+            Cost::ExtCost { ext_cost_kind: ExtCosts::promise_return } => &mut self.data[59],
+            Cost::ExtCost { ext_cost_kind: ExtCosts::validator_stake_base } => &mut self.data[60],
+            Cost::ExtCost { ext_cost_kind: ExtCosts::validator_total_stake_base } => {
+                &mut self.data[61]
+            }
+            #[cfg(feature = "protocol_feature_alt_bn128")]
+            Cost::ExtCost { ext_cost_kind: ExtCosts::alt_bn128_g1_multiexp_base } => {
+                &mut self.data[62]
+            }
+            #[cfg(feature = "protocol_feature_alt_bn128")]
+            Cost::ExtCost { ext_cost_kind: ExtCosts::alt_bn128_g1_multiexp_byte } => {
+                &mut self.data[63]
+            }
+            #[cfg(feature = "protocol_feature_alt_bn128")]
+            Cost::ExtCost { ext_cost_kind: ExtCosts::alt_bn128_g1_multiexp_sublinear } => {
+                &mut self.data[64]
+            }
+            #[cfg(feature = "protocol_feature_alt_bn128")]
+            Cost::ExtCost { ext_cost_kind: ExtCosts::alt_bn128_pairing_check_base } => {
+                &mut self.data[65]
+            }
+            #[cfg(feature = "protocol_feature_alt_bn128")]
+            Cost::ExtCost { ext_cost_kind: ExtCosts::alt_bn128_pairing_check_byte } => {
+                &mut self.data[66]
+            }
+            #[cfg(feature = "protocol_feature_alt_bn128")]
+            Cost::ExtCost { ext_cost_kind: ExtCosts::alt_bn128_g1_sum_base } => &mut self.data[67],
+            #[cfg(feature = "protocol_feature_alt_bn128")]
+            Cost::ExtCost { ext_cost_kind: ExtCosts::alt_bn128_g1_sum_byte } => &mut self.data[68],
+            Cost::ExtCost { ext_cost_kind: ExtCosts::__count } => unreachable!(),
+        }
+    }
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, PartialEq, Clone, Eq, Debug)]
+pub struct CostGasUsed {
+    #[serde(flatten)]
```

Comment:
Let's keep the bits related to JSON serilization in the views.rs file. profile.rs doesn't need to care what the public JSON API looks like.

Reply:
OK!